app/models/enhancements/artefact.rb

```diff
@@ -9,24 +9,24 @@ class Artefact
   # When saving an artefact we want to send it to the router.
   after_save :update_router
 
+  has_and_belongs_to_many :related_artefacts, class_name: "Artefact"
```

Comment:
These additions threw me a little - feel like they warrant their own commit with some explanation of what's going on here.